lib/app/dispatcher.rb

```diff
@@ -7,31 +7,32 @@
 # frozen_string_literal: true
 
 require 'active_support'
-require 'concurrent-ruby'
+require 'concurrent'
 require 'connectors'
 require 'core'
 require 'utility'
 require 'app/config'
 
 module App
   class Dispatcher
-    POLL_IDLING = (App::Config[:idle_timeout] || 60).to_i
-    TERMINATION_TIMEOUT = (App::Config[:termination_timeout] || 60).to_i
-    MIN_THREADS = (App::Config.thread_pool&.min_threads || 0).to_i
-    MAX_THREADS = (App::Config.thread_pool&.max_threads || 5).to_i
-    MAX_QUEUE = (App::Config.thread_pool&.max_queue || 100).to_i
+    POLL_INTERVAL = (App::Config.poll_interval || 3).to_i
+    TERMINATION_TIMEOUT = (App::Config.termination_timeout || 60).to_i
+    HEARTBEAT_INTERVAL = (App::Config.heartbeat_interval || 60 * 30).to_i
+    MIN_THREADS = (App::Config.dig(:thread_pool, :min_threads) || 0).to_i
```

Comment:
What'll happen when we have 0 threads? Is this a valid setting at all? If yes, what will happen when something is submitted to this pool?
TLDR: maybe we should default this to 1?

Reply:
>TLDR: maybe we should default this to 1?

The default will only work if it's not configured in YAML.

I tried to add validation (e.g. when both `min_threads` and `max_threads` are available, `max_threads` must not be less than `min_threads`) but it doesn't work. There are other validations needed, e.g. when it's non-native mode, both `connector_id` and `service_type` must be configured. I have created an issue https://github.com/elastic/enterprise-search-team/issues/2749 to track this. cc @elastic/ingestion-team